internal/controllers/clusterclass/clusterclass_controller.go

```diff
@@ -263,3 +326,52 @@ func (r *Reconciler) reconcileExternal(ctx context.Context, clusterClass *cluste
 func uniqueObjectRefKey(ref *corev1.ObjectReference) string {
 	return fmt.Sprintf("Name:%s, Namespace:%s, Kind:%s, APIVersion:%s", ref.Name, ref.Namespace, ref.Kind, ref.APIVersion)
 }
+
+// extensionConfigToClusterClass maps an ExtensionConfigs with the corresponding ClusterClass to reconcile them on updates
+// of the ExtensionConfig.
+func (r *Reconciler) extensionConfigToClusterClass(o client.Object) []reconcile.Request {
+	res := []ctrl.Request{}
+
+	ext, ok := o.(*runtimev1.ExtensionConfig)
+	if !ok {
+		panic(fmt.Sprintf("Expected an ExtensionConfig but got a %T", o))
+	}
+
+	clusterClasses := clusterv1.ClusterClassList{}
+	selector, err := metav1.LabelSelectorAsSelector(ext.Spec.NamespaceSelector)
+	if err != nil {
+		return nil
+	}
+	if err := r.Client.List(context.TODO(), &clusterClasses); err != nil {
+		return nil
+	}
+	for _, clusterClass := range clusterClasses.Items {
+		for _, patch := range clusterClass.Spec.Patches {
+			if patch.External != nil && patch.External.DiscoverVariablesExtension != nil {
+				if matchNamespace(context.TODO(), r.Client, selector, clusterClass.Namespace) {
```

Comment:
Can we move this into the outer for loop (l.348) and do a continue if the namespaces doesn't match?